src/ai/ask.rs

```diff
@@ -83,12 +84,13 @@ pub async fn prepare_ask_req(
             break;
         }
         context.push(doc.content);
-        refs.push(RefDoc {
+        ref_set.insert(RefDoc {
```

Comment:
@caugner 
```rust
        if refs.iter().any(|r: &RefDoc| r.slug == doc.slug) {
            refs.push(RefDoc {
                url: doc.url,
                slug: doc.slug,
                title: doc.title,
            });
        }
```
That would have been my initial approach as it's not to many (only 3 right now).

Using a set is fine, but it this case maybe we wanna keep the order and use a [BTreeSet](https://doc.rust-lang.org/std/collections/struct.BTreeSet.html)

Reply:
[6fdc8e5](https://github.com/mdn/rumba/pull/230/commits/6fdc8e5aae37c80dc2ff4d3d2538b7fc1dc7f861)